omics/uriparse/uri_parse.py

```diff
@@ -14,45 +14,30 @@ class OmicsUri:
     ex: omics://429915189008.storage.us-east-1.amazonaws.com/1981413158/readSet/5346184667/source1
     ex: omics://429915189008.storage.us-east-1.amazonaws.com/1981413158/reference/5346184667/source
     """
-
-    SCHEME = "omics"
-    NETLOC = "STORAGE"
-    DEFAULT_FILE_NAME = None
+    URI_REGEX = r"omics://(\d{10,12})\.storage\.([a-z]{2}-[a-z-]{4,}-\d+)\.amazonaws\.com/(\d{10,36})/(readSet|reference)/(\d{10,36})(/(source[12]?|index))?$"
 
     def __init__(self, omics_uri):
         """Initialize an Omics URI.
 
         Args:
             omics_uri: String representing an omics URI.
         """
-        # https://docs.python.org/3/library/urllib.parse.html
-        url_parts = urlparse(omics_uri.upper())
-        if url_parts.scheme != self.SCHEME:
-            raise ValueError(f"Invalid URI scheme, expected {self.SCHEME}")
-
-        if self.NETLOC not in url_parts.netloc:
-            raise ValueError(f"Invalid URI netloc, expected {self.NETLOC}")
-
-        self._uri_path = url_parts.path
-
-        (
-            self.store_id,
-            self.resource_type,
-            self.resource_id,
-            self.file_name,
-            *_,
-        ) = self._uri_path.strip("/").split("/") + [self.DEFAULT_FILE_NAME, None]
-
-        if self.resource_type not in OmicsFileType.list():
-            raise ValueError(
-                f"Invalid URI path for reference, {self.resource_type} is not a valid URI type"
-            )
-
-        if (
-            self.file_name
-            not in OMICS_URI_TYPE_FILENAME_MAP[OmicsFileType(self.resource_type)].list()
-        ):
-            raise TypeError("URI file is unsupported.")
+        uri_match = re.match(self.URI_REGEX, omics_uri, re.IGNORECASE)
```

Comment:
So much better than splitting the string!  :)  Thanks for making the switch to regex.